File: add/codesnippet/VisualBasic/4a8f6836-134b-46d5-88fc-_1.vb
    Structure Pet
        Public Name As String
        Public Vaccinated As Boolean
    End Structure

    Shared Sub CountEx2()
        ' Create an array of Pet objects.
        Dim pets() As Pet = {New Pet With {.Name = "Barley", .Vaccinated = True}, _
                       New Pet With {.Name = "Boots", .Vaccinated = False}, _
                       New Pet With {.Name = "Whiskers", .Vaccinated = False}}

        ' Count the number of unvaccinated pets in the array.
        Dim numberUnvaccinated As Integer = pets.AsQueryable().Count(Function(p) p.Vaccinated = False)

        MsgBox(String.Format("There are {0} unvaccinated animals.", numberUnvaccinated))
    End Sub

    ' This code produces the following output:
    '
    ' There are 2 unvaccinated animals.

File: add/codesnippet/CSharp/4a8f6836-134b-46d5-88fc-_1.cs
            class Pet
            {
                public string Name { get; set; }
                public bool Vaccinated { get; set; }
            }

            public static void CountEx2()
            {
                // Create an array of Pet objects.
                Pet[] pets = { new Pet { Name="Barley", Vaccinated=true },
                               new Pet { Name="Boots", Vaccinated=false },
                               new Pet { Name="Whiskers", Vaccinated=false } };

                // Count the number of unvaccinated pets in the array.
                int numberUnvaccinated =
                    pets.AsQueryable().Count(p => p.Vaccinated == false);

                Console.WriteLine(
                    "There are {0} unvaccinated animals.",
                    numberUnvaccinated);
            }

            // This code produces the following output:
            //
            // There are 2 unvaccinated animals.
